src/openqaoa-core/algorithms/qaoa/qaoa_workflow.py

```diff
@@ -190,7 +289,17 @@ def compile(self, problem: QUBO = None, verbose: bool = False):
         verbose: bool
             Set True to have a summary of QAOA to displayed after compilation
         """
-
+        # if isinstance(routing_function,Callable):
+        #     #assert that routing_function is supported only for Standard QAOA.
+        #     if (
+        #         self.backend_properties.append_state is not None or\
+        #         self.backend_properties.prepend_state is not None or\
+        #         self.circuit_properties.mixer_hamiltonian is not 'x' or\
+                
+        #     )
+
+        # connect to the QPU specified
+        self.device.check_connection()
```

Comment:
Do we need this line here. We already do check_connection in the Backends.

Reply:
If the device is not instantiated properly before the routing_function is called, it returns an error. This is because we need to connect to the cloud devices before we pull their coupling_maps